torch/_functorch/partitioners.py

```diff
@@ -474,7 +452,8 @@ def insert_node_in_graph(node):
             if order[user] < minimum_order:
                 minimum_order = order[user]
                 first_node_in_bwd = user
-    assert first_node_in_bwd is not None
+    if first_node_in_bwd is None:
+        return gm
```

Comment:
why is this ?

Reply:
Fixes a bug in existing impl - if gradInp does not depend on gradOut then this will not detect any nodes in the bacwkards.

Added a comment.